src/Java.Interop/Java.Interop/JniRuntime.JniValueManager.cs

```diff
@@ -598,19 +609,19 @@ public override Expression CreateParameterFromManagedExpression (JniValueMarshal
 						test:       Expression.Equal (Expression.Constant (null), sourceValue),
 						ifTrue:     Expression.Assign (r, Expression.New (typeof (JniObjectReference))),
 						ifFalse:    Expression.Assign (r, Expression.Property (sourceValue, "PeerReference"))));
-			context.CleanupStatements.Add (DisposeObjectReference (r));
 
-			var h   = Expression.Variable (typeof (IntPtr), sourceValue + "_handle");
-			context.LocalVariables.Add (h);
-			context.CreationStatements.Add (Expression.Assign (h, Expression.Property (r, "Handle")));
-			return h;
+			if (typeof (IJavaPeerable).GetTypeInfo ().IsAssignableFrom (sourceValue.Type.GetTypeInfo ())) {
```

Comment:
This condition should always be true: this is within `JavaPeerableValueMarshaler` -- a `JniValueMarshaler<IJavaPeerable>` -- so `sourceValue` *had better be* an `IJavaPeerable` or *something* has gotten very horribly confused.  (Probably me!)

Reply:
Yeah, that just me being too careful :-) I guess we can drop the test as it will fail in the next line pretty clearly in case something got broken horribly.